src/Database/PostgreSQL/Simple/Migration.hs

```diff
@@ -157,10 +158,24 @@ initializeSchema con verbose = do
         [ "create table if not exists schema_migrations "
         , "( filename varchar(512) not null"
         , ", checksum varchar(32) not null"
-        , ", executed_at timestamp without time zone not null default now() "
+        , ", executed_at timestamp with time zone not null default now() "
         , ");"
         ]
 
+-- | Migrates the helper table in case it was created by an older version of
+-- initializeSchema.
+migrateMeta :: Connection -> Bool -> IO ()
+migrateMeta con verbose = do
+    executedAtType <- getDataType con "schema_migrations" "executed_at"
+    when (executedAtType == Just "timestamp without time zone") $ do
```

Comment:
How is user data corrupted by this patch?  The only thing done that is new behavior is triggered only in that one specific `Just` case.  Are you talking about possible corruption that would also happen without this patch, but you are saying that this would be a good place to prevent it now that we have a place to check the data type?